src/sentry/api/event_search.py

```diff
@@ -2154,17 +2173,22 @@ def result_type_fn(function_arguments, parameter_values):
                         ],
                     ],
                 ],
-                "count_range_{index:g}",
+                "{query_alias}",
             ],
             default_result_type="integer",
         ),
         Function(
             "percentage",
-            required_args=[FunctionArg("numerator"), FunctionArg("denominator")],
+            required_args=[
+                FunctionArg("numerator"),
+                FunctionArg("denominator"),
+                FunctionArg("query_alias"),
+            ],
+            # aggregate and not a column since it needs to act on aggregates itself
```

Comment:
I don't think this is the reason why `percentage` is defined as an aggregate here. I believe the reason is that you need to pass the expression through to clickhouse as is without modifications. And defining it as a column does not let you do that at this time.

Reply:
So during exploration of this (and a few other solutions) I tried flipping this to a column, which then meant that conditions on it ended up in the `WHERE` which would break since the conditions that we're using this with contain aggregates, and have to end up in the `HAVING`